### DISCUSSION TITLE
Adding Authentication

### DISCUSSION THREAD

**Discussion by veerbal1** [EddieHubCommunity/LinkFree#295] [Ideas]
Can we implement an authentication system in which user can sign up, and fill details at his own.
And our system generate a link, that will show up all user's social media details.

and Then things will become automatic.

**Comment by gokulrajpuviyarasu** [comment]
Its too complicated @veerbal1

**Comment by eddiejaoude** [comment]
That is something we could do later, I think these are the steps we need to do

1a. Add commands to EddieBot for people to update their information
1b. Have LinkFree read the data from the API
2. Add OAuth authentication to the LinkFree and the API

**Reply by loftwah** [reply]
Basic Auth is the best auth :D Although in this case, OAuth is definitely suitable.

**Comment by eddiejaoude** [comment]
This has now be done so I will close
